projects/pastanaga-angular/src/lib/toast/toast.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+
+@Component({
+    template: ``,
+})
+export class TestComponent {
+    constructor(private service: ToastService) {}
+}
+
+describe('ToastService', () => {
+    let nextId = 0;
+    const createService = createComponentFactory({
+        component: TestComponent,
+        overrideModules: [
+            [
+                BrowserDynamicTestingModule,
+                {
+                    set: {
+                        imports: [PaToastModule],
+                        entryComponents: [ToastComponent],
+                    },
+                },
+            ],
+        ],
+    });
+    let spectator: Spectator<TestComponent>;
+    let service: ToastService;
+    let appRef: ApplicationRef;
+    let toastStatus: BehaviorSubject<ToastStatus>;
+    let renderer: Renderer2;
+    let toastContainer: HTMLElement | undefined;
+    let toastMap: Map<string, ComponentRef<ToastComponent>>;
+    beforeEach(() => {
+        spectator = createService();
+        service = spectator.inject(ToastService);
+        appRef = spectator.inject(ApplicationRef);
+        // @ts-ignore accessing private member
+        toastStatus = service._toastStatus;
+        // @ts-ignore accessing private member
+        renderer = service._renderer;
+        // @ts-ignore accessing private member
+        toastMap = service._toastMap;
+    });
+
+    afterEach(() => {
+        // @ts-ignore accessing private member
+        service._toastMap.clear();
+    });
+
+    it('should open a toast without config', () => {
+        const attachView = jest.spyOn(appRef, 'attachView');
+        const setAttributes = jest.spyOn(renderer, 'setAttribute');
+        const appendChild = jest.spyOn(renderer, 'appendChild');
+        service.open('a message', 'success');
+
+        expect(toastMap.size).toEqual(1);
+        const createdToast: ComponentRef<ToastComponent> | undefined = toastMap.get(`pa-toast-${nextId}`);
+        expect(createdToast).toBeTruthy();
+
+        expect(attachView).toHaveBeenCalledWith(createdToast?.hostView);
+        nextId++;
+        // @ts-ignore accessing private member
+        toastContainer = service._toastContainer;
+
+        expect(toastContainer).toBeTruthy();
+
+        expect(setAttributes).toHaveBeenCalledWith(createdToast?.location.nativeElement, 'role', 'alert');
+        expect(appendChild).toHaveBeenCalledWith(toastContainer, createdToast?.location.nativeElement);
+
+        expect(toastStatus.value).toEqual('opened');
+    });
+
+    it('should open a toast with config', () => {
+        service.open('a message', 'success', { buttonLabel: 'label' });
+        expect(toastMap.size).toEqual(1);
+        const createdToast: ComponentRef<ToastComponent> | undefined = toastMap.get(`pa-toast-${nextId}`);
+        expect(createdToast).toBeTruthy();
+        nextId++;
+        // @ts-ignore accessing private member
+        expect(createdToast?.instance._actionButtonLabel).toEqual('label');
+    });
+
+    it('should open an info toast', () => {
```

Comment:
for all these `openInfo`, `openSuccess` methods, just check they call `open` with the right parameters. 
you then test `open` method on its own.

Reply:
it feels more resilient to me. if the implementation change, we still need the openInfo to produce an Info toast

Reply:
if the implementation change and you don't call open method anymore, then the test would fail.
here you're testing several times the same method: `open`. I don't think it's good for test performances